src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -150,12 +176,19 @@ private IntPtr GetPin(Object obj)
             }
             return handle.AddrOfPinnedObject();
         }
-        private void FlushPins()
+
+        private List<IntPtr> _nativeExceptions = new List<IntPtr>();
```

Comment:
This list should be created lazily - only once there is exception to store into it.